mesonbuild/interpreter/primitives/string.py

```diff
@@ -104,6 +105,18 @@ def arg_replace(match: T.Match[str]) -> str:
 
         return re.sub(r'@(\d+)@', arg_replace, self.held_object)
 
+    @noKwargs
+    @noPosargs
+    def parse_as_list_method(self, args: T.List[TYPE_var], kwargs: TYPE_kwargs) -> T.List[str]:
+        lines = self.held_object.splitlines()
+        parts = []
+        for line in lines:
+            line = line.strip()
+            if line == '':
```

Comment:
w.r.t. the originally stated use case, is it actually expected for the general use case to strip leading/trailing whitespace when handling a file that was read as a list of filenames or other newline-delimited tokens? Or is the splitlines on its own sufficient?

Reply:
The way I see it, there are two questions: 1) do we want to support paths/entries with leading/trailing whitespace, and 2) do we want to be able to represent empty entries?

I decided to not support leading/trailing whitespace, since the `strip().split('\n')` approach also doesn't really support that. But on second thought, that might've been the wrong call, and the `line = line.strip()` should probably be removed.

The question of whether to support empty entries is more difficult IMO. But I think people are likely to want to separate chunks of entries with whitespace if they're maintaining a list file by hand. However, for the use-case of hand-editing a file, people would probably want comments and such as well, which seems pretty out of scope.

I think I've been thinking of this problem a bit too much as designing a list specification language, when a nicer `split('\n')` is probably more appropriate. I think the method should be renamed to `splitlines()` and just `return self.helf_object.splitlines()`. I'll push an update if you agree.